doc/api/n-api.md

```diff
@@ -3134,6 +3134,35 @@ overhead in creating/storing strings with this method.
 The JavaScript `string` type is described in
 [Section 6.1.4][] of the ECMAScript Language Specification.
 
+#### `napi_set_named_property_len`
+
+<!-- YAML
+added:
+  - v22.1.0
+-->
+
+> **Stability**: 1 - Experimental
+
+```c
+napi_status NAPI_CDECL napi_set_named_property_len(napi_env env,
+                                                   napi_value object,
+                                                   const char* utf8name,
+                                                   size_t name_length,
+                                                   napi_value value);
+```
+
+* `[in] env`: The environment that the API is invoked under.
+* `[in] object`: The JavaScript object on which to set the named property.
+* `[in] utf8name`: Character buffer representing a UTF-8-encoded string.
+* `[in] name_length`: The length of the string in bytes, or NAPI\_AUTO\_LENGTH if it is null-terminated.
+* `[in] value`: The napi\_value representing the value to set as the named property.
+  Returns napi\_ok if the API succeeded.
+
+This API sets a named property on a JavaScript object, treating '\0'
+characters as values rather than terminators. This allows for more
+flexible property names, especially when dealing with virtual module
+names or other scenarios where '\0' characters are used as part of the name.
```

Comment:
```suggestion
characters as values rather than terminators.
```

Do we need to point out specific use cases? Perhaps we should concentrate on documenting the new capability itself.